koocook_core/views/recipes.py

```diff
@@ -91,3 +98,13 @@ def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         context['ingredients'] = json.dumps([ing.to_dict for ing in list(self.get_object().recipe_ingredients.all())])
         return context
+
+
+class RecipeDetailView(DetailView):
+    model = Recipe
+    template_name = 'recipes/detail.html'
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context['ingredients'] = [ing.to_dict for ing in list(self.get_object().recipe_ingredients.all())]
```

Comment:
I've said this on pull request #8 

>These look more correct as methods by their names, but OK I guess

_Originally posted by @MaiMee1 in https://github.com/_render_node/MDIzOlB1bGxSZXF1ZXN0UmV2aWV3VGhyZWFkMjEyMTIzODMzOnYy/pull_request_review_threads/discussion_

I don't think this is OK anymore. Now we have both `as_dict` as a property for `SerialisableModel` (which I also just noticed that it is not in American English, which should also be changed some time in the future or, better, now) and `to_dict` in other models. I think it is confusing. Not to mention how `as_sthsth` is normally implemented as a function, as seen in many other repos and in the standard library. To quote a few, there's [`pathlib.PurePath.as_posix()`](https://docs.python.org/3.8/library/pathlib.html#pathlib.PurePath.as_posix), [`pathlib.PurePath.as_uri()`](https://docs.python.org/3.8/library/pathlib.html#pathlib.PurePath.as_uri), [`decimal.Decimal.as_tuple()`](https://docs.python.org/3.8/library/decimal.html#decimal.Decimal.as_tuple), [`email.message.EmailMessage.as_bytes()`](https://docs.python.org/3.8/library/email.message.html#email.message.EmailMessage.as_bytes)and many more. I think this needs to be changed along with `SerialisableModel.as_dict` and `SerialisableModel.as_json` in `koocook_core/models/base.py`.

Reply:
I already noticed them. I left them as is, though, because it had been part of many codebases in many branches. There will be a **huge** conflict resolution if this is done now.

Reply:
Still, there will be a fix for them soon.

Reply:
I found a reason for this. When passing a Django context into a template, Django will invoke the property **as if they were a non-callable variable** or `property`.

Reply:
I believe you can also call functions? You just can't pass it any arguments

Reply:
Here is an example https://docs.djangoproject.com/en/2.2/ref/templates/language/#accessing-method-calls

Reply:
It does invoke functions, but Django returns functions themselves, not the return parameter.

Reply:
This will definitely be fixed after all the data are retrieved via Ajax.

Reply:
Conventionally, your statement is right. Do you really want me to apply this convention? This will break the behaviour of posts  since I fixed that in another branch.

Reply:
> It does invoke functions, but Django returns functions themselves when serializing json data into the context instead of the return parameter.

I don't understand what you mean

Reply:
> Conventionally, your statement is right. Do you really want me to apply this convention? This will break the behaviour of posts since I fixed that in another branch.

Is this issue already fixed in another branch? what branch? If not I think fixing it now (and thereby also changing the code in posts) is better. No need to wait to fix things. We might get more dependencies and can't fix anything in the end.

Reply:
Since you say this is already solved in another branch `feature/rating` I'll close this. But be sure to fix this.